### PR TITLE
authentication via oauth2, issue #170

### PR DESCRIPTION
**Pull request by stefanlack**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -166,6 +179,13 @@ dependencies {
 
 }
 
+//dependencyManagement {
+//    imports {
+//        mavenBom "org.keycloak.bom:keycloak-adapter-bom:${keycloakVersion}"
+//    }
+//}
+
+
```
Comment (tjaeschke) [inline build.gradle]: This section should be removed

File: src/test/java/org/opendevstack/provision/controller/E2EProjectAPIControllerTest.java [hunk 2]
```diff
@@ -1,587 +1,834 @@
...
+  @InjectMocks @Autowired private ConfluenceAdapter realConfluenceAdapter;
+
+  @InjectMocks @Autowired private BitbucketAdapter realBitbucketAdapter;
+
+  @InjectMocks @Autowired private RundeckAdapter realRundeckAdapter;
+
+  @Mock RestClient restClient;
+  RestClientMockHelper mockHelper;
+
+  // @Mock OldRestClient mockOldRestClient;
```
Comment (tjaeschke) [inline src/test/java/org/opendevstack/provision/controller/E2EProjectAPIControllerTest.java]: Is this still needed?

File: src/test/java/org/opendevstack/provision/controller/E2EProjectAPIControllerTest.java [hunk 3]
```diff
@@ -1,587 +1,834 @@
...
+    }
+
+    // rundeck create-projects job execution
+    ExecutionsData execution =
+        readTestData("rundeck-create-project-response", ExecutionsData.class);
+
+    String createJobId = realJobStore.getJobIdForJobName("create-projects");
+    assertNotNull(createJobId);
+
+    //    Mockito.when(
```
Comment (tjaeschke) [inline src/test/java/org/opendevstack/provision/controller/E2EProjectAPIControllerTest.java]: Is the call still necessary?

File: src/test/java/org/opendevstack/provision/services/BitbucketAdapterTest.java [hunk 4]
```diff
@@ -302,22 +322,26 @@ public void callCreateRepoApiTest() throws Exception {
...
+        .setRepositoryPermissions(eq(expected), eq(projectKey), eq("groups"), any());
 
-    Mockito.verify(spyAdapter).setRepositoryPermissions(eq(expected), eq(projectKey), eq("users"),
-        any());
+    Mockito.verify(spyAdapter)
+        .setRepositoryPermissions(eq(expected), eq(projectKey), eq("users"), any());
 
-    Mockito.verify(client).callHttp(eq(uri), eq(repo), anyBoolean(), eq(RestClient.HTTP_VERB.POST),
-        any());
+    //    Mockito.verify(restClient)
```
Comment (tjaeschke) [inline src/test/java/org/opendevstack/provision/services/BitbucketAdapterTest.java]: And the rest client question. ;-)

File: src/test/java/org/opendevstack/provision/controller/E2EProjectAPIControllerTest.java (hunk 2 above)
Comment (stefanlack) [reply src/test/java/org/opendevstack/provision/controller/E2EProjectAPIControllerTest.java]: I will remove this commented out obsolte code. In this class there is still one test ignored and I will fix this test.

**Comment by stefanlack** [comment]
@clemensutschig yes, we need a documentation. I am going to add a documentation on monday.

File: build.gradle (hunk 1 above)
Comment (stefanlack) [reply build.gradle]: Ok, please see my last commits on this PR.

File: src/test/java/org/opendevstack/provision/controller/E2EProjectAPIControllerTest.java (hunk 3 above)
Comment (stefanlack) [reply src/test/java/org/opendevstack/provision/controller/E2EProjectAPIControllerTest.java]: Ok, please see my last commits on this PR.

File: src/test/java/org/opendevstack/provision/services/BitbucketAdapterTest.java (hunk 4 above)
Comment (stefanlack) [reply src/test/java/org/opendevstack/provision/services/BitbucketAdapterTest.java]: Ok, please see my last commits on this PR.

**Pull request merged**